File: leetcode/python/passedProblems/7-reverse-integer.py
#Reverse digits of an integer.

#Example1: x = 123, return 321
#Example2: x = -123, return -321 

class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        re = 0;
        minus = (x >= 0 and 1 or 0);
        #print minus
        #print x, ",", (x>=0)
        x = x > 0 and x or -x;
        while x != 0:
            res = x % 10;
            re = re * 10 + res;
            x /= 10;
        re = minus == 0 and -re or re;
        re = 0 if (re > 2147483647 or re < - 2147483648) else re ;
        return re;

s = Solution();
#print s.reverse(123);
print s.reverse(1534236469);
#print s.reverse(-123);


        

File: leetcode/cpp/7-reverse-integer.cpp
/*
 * 颠倒数字
Example1: x = 123, return 321
Example2: x = -123, return -321

 */


class Solution {
public:
    bool myLE(string &a, string &b){
        if(a.length() != b.length()) return a.length() < b.length();
        for(int i = 0  ; i < a.length(); i ++)
            if(a[i]!= b[i])
                return a[i] < b[i];
        return true;
    }
    int reverse(int x) {
        bool neg = x < 0;
        if(neg) x = -x;
        string t(to_string(x));
        std::reverse(t.begin(), t.end());
        string mxt(to_string(INT_MAX));
        if(myLE(mxt, t)) return 0;
        int re = stoi(t);
        if(neg) re = -re;
        return re;
    }
};
